omnibus/files/private-chef-cookbooks/private-chef/spec/libraries/private_chef_spec.rb

```diff
@@ -0,0 +1,199 @@
+      expect(Chef::Log).to receive(:fatal)
+      expect {config_for("backend-passive.chef.io")}.to raise_error SystemExit
+    end
+
+    it "sets the relavent data directory paths to the shared store" do
+      rendered_config = config_for("backend-active.chef.io")
+      expect(rendered_config["private_chef"]["bookshelf"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/bookshelf")
+      expect(rendered_config["private_chef"]["rabbitmq"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/rabbitmq")
+      expect(rendered_config["private_chef"]["opscode-solr4"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/opscode-solr4")
+      expect(rendered_config["private_chef"]["redis_lb"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/redis_lb")
+
+      # Legacy Paths
+      expect(rendered_config["private_chef"]["couchdb"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/couchdb")
+      expect(rendered_config["private_chef"]["opscode-solr"]["data_dir"]).to eq("/var/opt/opscode/drbd/data/opscode-solr")
+      expect(rendered_config["private_chef"]["opscode-chef"]["checksum_path"]).to eq("/var/opt/opscode/drbd/data/opscode-chef/checksum")
+    end
+
+    it "sets backend services to listen on INADDR_ANY if the machine is a backend" do
+      rendered_config = config_for("backend-active.chef.io")
+      expect(rendered_config["private_chef"]["rabbitmq"]["node_ip_address"]).to eq("0.0.0.0")
+      expect(rendered_config["private_chef"]["bookshelf"]["listen"]).to eq("0.0.0.0")
+      expect(rendered_config["private_chef"]["redis_lb"]["listen"]).to eq("0.0.0.0")
+      expect(rendered_config["private_chef"]["opscode-solr4"]["ip_address"]).to eq("0.0.0.0")
+      expect(rendered_config["private_chef"]["postgresql"]["listen_address"]).to eq("*")
+    end
+
+    it "sets the VIPs for backend services to the backend_vip when configuring frontend services" do
+      expect_existing_secrets
+      rendered_config = config_for("frontend.chef.io")
+      expect(rendered_config["private_chef"]["postgresql"]["vip"]).to eq("10.0.0.1")
+      expect(rendered_config["private_chef"]["bookshelf"]["vip"]).to eq("10.0.0.1")
+      expect(rendered_config["private_chef"]["redis_lb"]["vip"]).to eq("10.0.0.1")
+      expect(rendered_config["private_chef"]["rabbitmq"]["vip"]).to eq("10.0.0.1")
+      expect(rendered_config["private_chef"]["opscode-solr4"]["vip"]).to eq("10.0.0.1")
+    end
+
+    it "disables backend services on the frontend" do
+      expect_existing_secrets
+      rendered_config = config_for("frontend.chef.io")
+      expect(rendered_config["private_chef"]["postgresql"]["enable"]).to eq(false)
+      expect(rendered_config["private_chef"]["bookshelf"]["enable"]).to eq(false)
+      expect(rendered_config["private_chef"]["redis_lb"]["enable"]).to eq(false)
+      expect(rendered_config["private_chef"]["rabbitmq"]["enable"]).to eq(false)
+      expect(rendered_config["private_chef"]["opscode-solr4"]["enable"]).to eq(false)
+    end
```

Comment:
Its generally considered bad Rspec practice to have more than one assert per `it` block, but I could see how breaking these all out into their own in this instance would be annoying :) Could probably loop or something, but its not a big deal.